.github/workflows/test.yml

```diff
@@ -66,14 +58,34 @@ jobs:
         uses: actions/cache@v2
         with:
           path: .tox
-          key: tox-cache-${{ matrix.python-version }}-${{ matrix.package }}-${{ matrix.os }}-${{ hashFiles('tox.ini', 'dev-requirements.txt') }}-core
+          key: tox-cache-${{ env.RUN_MATRIX_COMBINATION }}-${{ hashFiles('tox.ini', 'dev-requirements.txt') }}-core
       - name: run tox
-        run: tox -f ${{ matrix.python-version }}-${{ matrix.package }}
+        run: tox -f ${{ matrix.python-version }}-${{ matrix.package }} -- --benchmark-json=${{ env.RUN_MATRIX_COMBINATION }}-benchmark.json
```

Comment:
I'm not too familiar with this syntax. What is this extra "-- --benchmark-json=${{ env.RUN_MATRIX_COMBINATION }}-benchmark.json" for?

Reply:
This just tells `pytest-benchmark` to output the results at this path. The path is relative to the current directory where `pytest` was run which is decided by tox in the `changedir` command. It's just a temporary file created during the test run so that we can pass it onto the `github-benchmark-action` github action later.

The detailed description is in the [pytest-benchmark command line arguments docs](https://pytest-benchmark.readthedocs.io/en/latest/usage.html#commandline-options)

> --benchmark-json=PATH
  | Dump a JSON report into PATH. Note that this will include the complete data (all the timings, not just the stats).

Reply:
Yup! That's outlined [here in the tox docs](https://tox.readthedocs.io/en/latest/example/general.html).

Reply:
Nice that was the bit I was missing ^